ray-operator/controllers/ray/common/service_test.go

```diff
@@ -126,3 +126,20 @@ func TestBuildServiceForHeadPodWithAnnotations(t *testing.T) {
 		t.Fatalf("Expected `%v` but got `%v`", annotations, svc.ObjectMeta.Annotations)
 	}
 }
+
+func TestBuildServiceForHeadPodWithCustomLabel(t *testing.T) {
+	labels := make(map[string]string)
+	labels["key"] = "testvalue"
+
+	svc, err := BuildServiceForHeadPod(*instanceWithWrongSvc, labels, nil)
+	assert.Nil(t, err)
+
+	// Selector should not contain any custom label.
+	if _, ok := svc.Spec.Selector["key"]; ok {
+		t.Fatalf("Expected `%v` not to exist", "key")
+	}
+
+	if _, ok := svc.ObjectMeta.Labels["key"]; !ok {
+		t.Fatalf("Expected `%v`", "key")
+	}
```

Comment:
We can also verify the length of `svc.Spec.Selector` and `svc.ObjectMeta.Labels` here.